lib/spack/spack/cmd/common/arguments.py

```diff
@@ -322,11 +322,68 @@ def add_cdash_args(subparser, add_help):
     )
 
 
-@arg
-def reuse():
-    return Args(
-        '--reuse', action='store_true', default=False,
-        help='reuse installed dependencies'
+class ConfigSetAction(argparse.Action):
+    """Generic action for setting spack config options from CLI.
+
+    This works like a ``store_const`` action but you can set the
+    ``dest`` to some Spack configuration path (like ``concretizer:reuse``)
+    and the ``const`` will be stored there using ``spack.config.set()``
+    """
+    def __init__(self,
+                 option_strings,
+                 dest,
+                 const,
+                 default=None,
+                 required=False,
+                 help=None,
+                 metavar=None):
+        # save the config option we're supposed to set
+        self.config_path = dest
+
+        # destination is translated to a legal python identifier by
+        # substituting '_' for ':'.
+        dest = dest.replace(":", "_")
+
+        super(ConfigSetAction, self).__init__(
+            option_strings=option_strings,
+            dest=dest,
+            nargs=0,
+            const=const,
+            default=default,
+            required=required,
+            help=help
+        )
```

Comment:
```suggestion
    def __init__(self,
                 option_strings,
                 config_path,
                 const,
                 default=None,
                 required=False,
                 help=None,
                 metavar=None):
        # save the config option we're supposed to set
        self.config_path = config_path

        super(ConfigSetAction, self).__init__(
            option_strings=option_strings,
            dest='fake',
            nargs=0,
            const=const,
            default=default,
            required=required,
            help=help
        )
```

I think this might be more legible to make clear we aren't using `dest`.

Reply:
I disagree w/this b/c now we have a variable called `fake` that nobody can query, because what's in it depends on how many of these config args you add.  The last one will win.  Having the variable saved into a unique location -- so that you can query later *whether* it was set from the command line -- is something users of this class may want to do.

Reply:
It's normal for argparse arguments to be available on the namespace that argparse produces - I don't see why this should be different (or handicapped).